### PR TITLE
Impersonation per server

### PR DESCRIPTION
**Pull request by frankgh**
Add tests for impersonation per server

### CODE REVIEW

**Comment by frankgh** [comment]
This PR needs to be merged after merging https://github.com/greenplum-db/pxf/pull/260

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/security/SecureLogin.java [hunk 1]
```diff
@@ -209,12 +209,13 @@ private LoginSession getServerLoginSession(final String serverName, final String
...
-        long kerberosMinMillisBeforeRelogin = Utilities.isSecurityEnabled(configuration) ?
+        boolean securityEnabled = Utilities.isSecurityEnabled(configuration);
+        long kerberosMinMillisBeforeRelogin = securityEnabled ?
                 PxfUserGroupInformation.getKerberosMinMillisBeforeRelogin(serverName, configuration) : 0L;
         LoginSession expectedLoginSession = new LoginSession(
                 configDirectory,
-                SecureLogin.getServicePrincipal(serverName, configuration),
-                SecureLogin.getServiceKeytab(serverName, configuration),
+                securityEnabled ? SecureLogin.getServicePrincipal(serverName, configuration) : null,
+                securityEnabled ? SecureLogin.getServiceKeytab(serverName, configuration) : null,
```
Comment (denalex) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/security/SecureLogin.java]: this has changed, you need to rebase off master

File: concourse/scripts/pxf_common.bash [hunk 2]
```diff
@@ -534,6 +534,19 @@ function configure_pxf_default_server() {
 			cp /etc/hbase/conf/*-site.xml "${PXF_CONF_DIR}/servers/default"
 		fi
 	fi
+
+	if [[ ${IMPERSONATION} == true ]]; then
+		cp -r ${PXF_CONF_DIR}/servers/default ${PXF_CONF_DIR}/servers/default-no-impersonation
+
+		if [[ ! -f ${PXF_CONF_DIR}/servers/default-no-impersonation/pxf-site.xml ]]; then
```
Comment (oliverralbertini) [inline concourse/scripts/pxf_common.bash]: What's the situation where this file would already be there?

If there isn't any, I suggest replacing this `if` and the `sed` call below with:
```bash
sed \
			-e "/<name>pxf.service.user.impersonation<\/name>/ {n;s|<value>.*</value>|<value>false</value>|g;}" \
			-e "/<name>pxf.service.user.name<\/name>/ {n;s|<value>.*</value>|<value>foobar</value>|g;}" \
			"${PXF_CONF_DIR}/templates/pxf-site.xml" > "${PXF_CONF_DIR}/servers/default-no-impersonation/pxf-site.xml"
```

File: concourse/scripts/test_pxf_multinode.bash [hunk 3]
```diff
@@ -111,6 +111,161 @@ function setup_pxf_on_cluster() {
...
+		if [[ ${IMPERSONATION} == true ]]; then
+			cp -r ${PXF_CONF_DIR}/servers/default ${PXF_CONF_DIR}/servers/default-no-impersonation
+
+			if [[ ! -f ${PXF_CONF_DIR}/servers/default-no-impersonation/pxf-site.xml ]]; then
+				cp ${PXF_CONF_DIR}/templates/pxf-site.xml ${PXF_CONF_DIR}/servers/default-no-impersonation/pxf-site.xml
+			fi
+			sed -i \
+			-e '/<name>pxf.service.user.impersonation<\/name>/ {n;s|<value>.*</value>|<value>false</value>|g;}' \
+			-e '/<name>pxf.service.user.name<\/name>/ {n;s|<value>.*</value>|<value>foobar</value>|g;}' \
+			${PXF_CONF_DIR}/servers/default-no-impersonation/pxf-site.xml
```
Comment (oliverralbertini) [inline concourse/scripts/test_pxf_multinode.bash]: This looks like the same code that's in `pxf_common.bash`. Might be worth refactoring to a function that lives there.

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/security/SecureLogin.java (hunk 1 above)
Comment (frankgh) [reply server/pxf-api/src/main/java/org/greenplum/pxf/api/security/SecureLogin.java]: yes!

File: concourse/scripts/pxf_common.bash (hunk 2 above)
Comment (frankgh) [reply concourse/scripts/pxf_common.bash]: depends on the job we are running, `pxf-site.xml` may or may not exist. We need to be careful not to overwrite it if it already exists .

File: concourse/scripts/test_pxf_multinode.bash (hunk 3 above)
Comment (frankgh) [reply concourse/scripts/test_pxf_multinode.bash]: the difference is that this code runs in an ssh block, not sure if it's possible to call a function in an ssh block if the function is not available in that block

Comment (oliverralbertini) [reply concourse/scripts/test_pxf_multinode.bash]: I would scp pxf_common.bash to the node and source pxf_common.bash in the ssh block. but feel free to just leave it as is, we can address this at a later time.

Comment (frankgh) [reply concourse/scripts/test_pxf_multinode.bash]: I expect to see this code refactored at a later point in the future :) most likely you will take care of make it into proper bash

**Pull request merged**